pkg/project/apiserver/admission/nodeenv/admission.go

```diff
@@ -31,9 +31,11 @@ type podNodeEnvironment struct {
 
 var _ = oadmission.WantsProjectCache(&podNodeEnvironment{})
 var _ = kadmission.WantsInternalKubeClientSet(&podNodeEnvironment{})
+var _ = admission.ValidationInterface(&podNodeEnvironment{})
+var _ = admission.MutationInterface(&podNodeEnvironment{})
 
 // Admit enforces that pod and its project node label selectors matches at least a node in the cluster.
-func (p *podNodeEnvironment) Admit(a admission.Attributes) (err error) {
+func (p *podNodeEnvironment) admit(a admission.Attributes, mutationAllowed bool) (err error) {
```

Comment:
The point of the boolean is to change behavior from mutation to validation only.
If you call the variable "validationOnly" then the code flow below can be changed from:
```
if mutationAllowed {
    ...
} else if (...) {
    return xyz
}
...
```
to a more readable:
```
if validationOnly {
    return xyz
}
... /* mutation and all */
```

At least I find this more readable and natural.

Reply:
We have this mutationAllowed style at a number of places. Would avoid to change that.

But I updated the code to follow your if-clause order, makes it easier, compare https://github.com/openshift/origin/pull/20743.